frontend/src/components/Calculator.js

```diff
@@ -25,81 +24,64 @@ import { onSnapshot, collection } from "firebase/firestore";
 
 import { scheduleInputsOnCalender } from "../utils/SchedulingLogic.js";
 
-import dayjs from "dayjs";
-
-const logoPath = "frontend/public/rocket_logo_full.png";
-const icon = <CheckBoxOutlineBlankIcon fontSize="small" />;
-const checkedIcon = <CheckBoxIcon fontSize="small" />;
-
-//for testing grid visualization
-// const Item = styled(Paper)(({ theme }) => ({
-//   backgroundColor: theme.palette.mode === 'dark' ? '#1A2027' : '#fff',
-//   ...theme.typography.body2,
-//   padding: theme.spacing(1),
-//   textAlign: 'center',
-//   color: theme.palette.text.secondary,
-// }));
-
-let calenderEvents = [];
-
 export default function Calculator() {
-  const [colleges, setColleges] = useState([]);
-
-  //console.log(colleges);
-
+  //Get school info from db
+  const [schoolInfo, setSchoolInfo] = useState([]);
   useEffect(
     () =>
       onSnapshot(collection(db, "colleges"), (snapshot) => {
-        setColleges(snapshot.docs.map((doc) => doc.data()));
+        setSchoolInfo(snapshot.docs.map((doc) => doc.data()));
       }),
     []
   );
 
-  function setChosenColleges(event, newSelectedColleges) {
-    console.log(selectedColleges);
-    setSelectedColleges(newSelectedColleges);
-    selectedColleges = newSelectedColleges; // removing this line might be necesarry later on if stuffs not adding up for this parameter
-    console.log(selectedColleges);
-  }
-
+  // UX VARIABLES/CONSTANTS
   const revsionChoices = Array.from({ length: 10 - 1 }, (x, i) => i + 2);
   const minCheckingDays = 5;
   const checkingLengthChoices = Array.from(
     { length: 10 - (minCheckingDays - 1) },
     (x, i) => i + minCheckingDays
   );
+  let calenderEvents = [];
 
   // USER INPUTS
-  var [selectedColleges, setSelectedColleges] = useState([]); // List of json objects [{deadline: , college: }]
+  let chosenColleges; // Colleges selected by user (Autocomplete is weird w/ useState)
   const [userWritingSpeed, setUserWritingSpeed] = useState([]); // Number days taken takes to write
 
   const [startDate, setStartDate] = useState(""); // do startDate.$d to access datetype
   const [checkingLength, setCheckingLength] = useState(""); // Number length of checking period in days
 
   const [revisionAmt, setRevisionAmt] = useState(""); // Number revisions on each essay (1 reivision per day)
 
-  // Handle Submit
+  // IMPORTANT FUNCTIONS
+  function setChosenColleges(newSelectedColleges) {
+    chosenColleges = newSelectedColleges;
+    console.log(chosenColleges);
+  }
 
   function handleSubmit() {
-    // console.log(calenderEvents);
-    console.log(selectedColleges);
-    for (var i = 0; i < selectedColleges.length; i++) {
-      selectedColleges[i].deadline = new Date(
-        selectedColleges[i].deadline.toDate().getTime()
+    if (chosenColleges) {
+      //Convert each date timestamp to date objects
```

Comment:
make these date conversions just happen when the colleges are imported from the db.

also add a loading thing if the colleges are undefined bc it takes a second to call from db